lib/matplotlib/figure.py

```diff
@@ -2240,24 +2250,32 @@ def __init__(self,
             %(Figure:kwdoc)s
         """
         super().__init__(**kwargs)
+        self._layout_engine = None
 
         if layout is not None:
-            if tight_layout is not None:
+            if (tight_layout is not None or
+                    constrained_layout is not None):
                 _api.warn_external(
-                    "The Figure parameters 'layout' and 'tight_layout' "
-                    "cannot be used together. Please use 'layout' only.")
+                    "The Figure parameters 'layout' and 'tight_layout' or "
+                    "'constrained_layout' cannot be used together. Please use "
+                    "'layout' only.")
```

Comment:
I'm not sure about squeezing these warnings together, the and/or can be confusing to parse.

Reply:
Done, though this whole block is a bit of a mess.  Do we have a redundant kwargs checker?